tajo-rpc/src/main/java/org/apache/tajo/rpc/AsyncRpcClient.java

```diff
@@ -113,16 +87,11 @@ public void close() {
   }
 
   private class ProxyRpcChannel implements RpcChannel {
-    private final ClientChannelInboundHandler handler;
-
-    public ProxyRpcChannel() {
-      this.handler = getChannel().pipeline()
-          .get(ClientChannelInboundHandler.class);
+    private ClientChannelInboundHandler handler;
 
-      if (handler == null) {
-        throw new IllegalArgumentException("Channel does not have " +
-            "proper handler");
-      }
+    private ClientChannelInboundHandler handler() {
+      return handler == null ? handler = getChannel().pipeline()
```

Comment:
getChannel() function has a possibility to return null when channelfuture is null, so it needs check null value.